src/tests/dbus-tests/force_clean.py

```diff
@@ -0,0 +1,76 @@
+import subprocess
+import syslog
+from blivet import Blivet
+
+
+class ForceClean(object):
+    """Class for force cleaning after tests have finished. The reason for this
+       is to ensure the same conditions for the next test run.
+       If everything works, this does nothing. Otherwise it informs about
+       actions performed and also writes them in the syslog."""
+
+    def __init__(self):
+        self.org_state_snapshot = None
+
+    @staticmethod
+    def _run_command(command):
+        res = subprocess.run(command, shell=True, stdout=subprocess.PIPE,
+                             stderr=subprocess.PIPE)
+        out = res.stdout.decode().strip()
+        return out
+
+    def _get_snapshot(self):
+        """Return current machine state - used for before
+           and after state comparison"""
+        result = self._run_command("lsblk -nlo NAME")
+        return result
+
+    def record_state(self):
+        """Get the first machine snapshot"""
+        self.org_state_snapshot = self._get_snapshot()
+
+    @staticmethod
+    def _get_diff(old_state, new_state):
+        old_list = old_state.split("\n")
+        new_list = new_state.split("\n")
+        diff = list(set(new_list) - set(old_list))
+        return diff
+
+    def restore_state(self):
+        """Get the second machine snapshot, compare it to the first one and perform
+           the cleaning. Return list of devices that could not be removed."""
+        new_state_snapshot = self._get_snapshot()
+        diff = self._get_diff(self.org_state_snapshot, new_state_snapshot)
+
+        if diff:
+            print("These devices were not properly cleaned by tests:\n" + "\n".join(diff))
+            print("Removing by force...")
+            # Put information into the syslog to be able to track possible issues
+            syslog.syslog("Following devices were not removed after Storaged D-Bus" +
+                          " tests and will be removed by force: %s" % str(diff)[1:-1])
+
+        for device in diff:
+            # kill all processes that are using the device
+            self._run_command("fuser -k /dev/%s" % device)
+            # umount device
+            self._run_command("umount /dev/%s" % device)
+            # detach device (will work if device is a loop, silently fails otherwise)
+            self._run_command("losetup --detach /dev/%s" % device)
+
+        self._run_command("modprobe -r scsi_debug")
+        self._run_command("targetcli clearconfig confirm=True")
+        blivet = Blivet()
+        # just try to remove everything
+        blivet.config.exclusive_disks = diff
+        blivet.reset()
+        blivet.devicetree.teardown_all()
```

Comment:
I think this should be moved above the commands are run and it will also do the work of the ``umount`` and ``losetup --detach`` above. Plus more things.